src/main/java/com/google/devtools/build/lib/bazel/bzlmod/IndexRegistry.java

```diff
@@ -237,14 +238,20 @@ private RepoSpec createArchiveRepoSpec(
       }
     }
 
-    return new ArchiveRepoSpecBuilder()
-        .setRepoName(repoName.getName())
-        .setUrls(urls.build())
-        .setIntegrity(sourceJson.get().integrity)
-        .setStripPrefix(Strings.nullToEmpty(sourceJson.get().stripPrefix))
-        .setRemotePatches(remotePatches.buildOrThrow())
-        .setRemotePatchStrip(sourceJson.get().patchStrip)
-        .build();
+    ArchiveRepoSpecBuilder archiveRepoSpecBuilder =
+        new ArchiveRepoSpecBuilder()
+            .setRepoName(repoName.getName())
+            .setUrls(urls.build())
+            .setIntegrity(sourceJson.get().integrity)
+            .setStripPrefix(Strings.nullToEmpty(sourceJson.get().stripPrefix))
+            .setRemotePatches(remotePatches.buildOrThrow())
+            .setRemotePatchStrip(sourceJson.get().patchStrip);
+
+    if (sourceJson.get().archiveType != null) {
+      archiveRepoSpecBuilder.setArchiveType(sourceJson.get().archiveType);
```

Comment:
fixed.

Reply:
hmm. this is breaking some tests.  Do we want a blank "type" always in the generated http_archive?  The previous way it would just omit it due to being null.  After this change it is ''.

We can keep the single builder call chain by using a ternary op and keeping the previous null behavior... perhaps this is better?

Reply:
you could change `ArchiveRepoSpecBuilder.setArchiveType(null)` to remove that key from the dict

Reply:
Hmm, I thought this shouldn't matter because the default value of `type` is `''`, see https://cs.opensource.google/bazel/bazel/+/master:src/main/java/com/google/devtools/build/lib/bazel/repository/starlark/StarlarkBaseExternalContext.java;l=564

But anyway, the current implementation looks good.

Reply:
That makes sense.  I could have updated the broken tests or set the default to '' instead of null.  I'll keep that in mind if I end up making another PR.  Thanks for the review and quick feedback!